bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/LspPlugin.java

```diff
@@ -42,7 +47,11 @@ public LspPlugin() {
 	public void start(BundleContext context) throws Exception {
 		super.start(context);
 		plugin = this;
-		cLanguageServerProvider = new CLanguageServerRegistry().createCLanguageServerProvider();
+		ServiceTracker<IWorkspace, IWorkspace> workspaceTracker = new ServiceTracker<>(context, IWorkspace.class, null);
+		workspaceTracker.open();
+		workspace = workspaceTracker.getService();
+		getCLanguageServerProvider();
```

Comment:
Here we have a mix of low level OSGi API with public accessors to stateful activator. It looks more consistent to use OSGi components for  `CLanguageServerRegistry` and `CompileCommandsDirLocatorRegistry`